app/connector/support/maven-plugin/src/main/java/io/syndesis/connector/support/maven/GenerateConnectorInspectionsMojo.java

```diff
@@ -80,7 +93,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 }
 
                 for (File file: files) {
-                    Connector connector = Json.reader().forType(Connector.class).readValue(file);
+                    JsonNode jsonNode = validateWithSchema(file.getName(), new FileInputStream(file));
```

Comment:
I would pass in `file` instead of `FileInputStream`, to make the responsibility of `validateWithSchema` to handle IO issues